### ISSUE TITLE
Best audio quality for youtube, Consensus is that it's m4a, when I tell your "Open Video Downloader"...

### ISSUE THREAD

**Issue by Intellectual234** [StefanLobbenmeier/youtube-dl-gui#30]
...to convert youtube videos to m4a, is it extracting the original m4a from the video/from youtube? or is it being inefficient & for example converting some other lower quality audio or OPUS to m4a?  m4a is always better than OPUS when it comes to youtube.

**Comment by StefanLobbenmeier** [comment]
Good question, i would have to research myself. You can turn on the setting [keep-unmerged-files](https://github.com/StefanLobbenmeier/youtube-dl-gui/wiki/Settings#keep-unmerged-files) to see which files are actually downloaded

**Comment by Intellectual234** [comment]
When i unmerge it for music.youtube.com & youtube.com, it downloads four file types oddly, f248.webm, f248.webm, mkv & OPUS. And the audio file connected to the mkv is OPUS also. How do i get ur program to obtain the original m4a file? I dont want it to convert OPUS to m4a. I want the original best quality m4a. My program "aTube Catcher" can obtain/rip the m4a, but i like your user interface/design much better.

Then again, its debated whether m4a or OPUS are different in quality. It probably depends of the bitrate?
Thing is, youtubedl terminal version allows you to download either m4a or OPUS, but your program doesnt.
See two links below, very helpful/educational. So here is what i think needs to be done, we need to get your program to download the m4a & the OPUS & then we need to compare the bitrate. -Thanks!

https://www.reddit.com/r/youtubedl/comments/p6myq4/deleted_by_user/

https://www.reddit.com/r/NewPipe/comments/nkhvpg/opus_vs_m4a/

**Comment by StefanLobbenmeier** [comment]
> youtubedl terminal version allows you to download either m4a or OPUS

Yeah this App something messed up with the format selection, see also #11.

so I think we can close this is the same swoop, when the format selection allows you to actually select the formats you want we can avoid issues like this.

**Comment by Intellectual234** [comment]
@StefanLobbenmeier , i dont not understand at all what you are trying to say, can you please clarify/elaborate? -Thanks

**Comment by StefanLobbenmeier** [comment]
Sorry i missed your message. In https://github.com/StefanLobbenmeier/youtube-dl-gui/issues/11#issuecomment-1374638970 I looked into the format selection.

Basically there is something already wrong with parsing the output of yt-dlp format selection. When this is fixed hopefully you can also select exactly the format you would like

**Comment by ZeldaTheSwordsman** [comment]
IMO you're honestly better off using the old Youtube-DL-GUI with an up-to-date yt-dlp executable.
